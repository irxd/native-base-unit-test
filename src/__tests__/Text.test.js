import React from 'react';
import { render } from '@testing-library/react-native';
import {
  NativeBaseProvider, Text
} from 'native-base';

describe('<Text />', () => {
  it('should render text component properly', () => {
    const { queryByTestId } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <Text testID="my-text">hello world</Text>
      </NativeBaseProvider>
    )

    expect(queryByTestId('my-text')).not.toBeNull()
  });

  it('should render text props properly', () => {
    const { queryByTestId } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <Text testID="my-text" fontWeight="bold" color="white">hello world</Text>
      </NativeBaseProvider>
    )
    
    expect(queryByTestId('my-text').props.style.fontWeight).toMatch('700');
    expect(queryByTestId('my-text').props.style.color).toMatch('#FFFFFF');
  });
})
