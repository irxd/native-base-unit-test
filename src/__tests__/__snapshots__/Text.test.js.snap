// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render text props properly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": undefined,
        "color": "#FFFFFF",
        "fontFamily": undefined,
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "700",
        "letterSpacing": 0,
        "lineHeight": 21,
        "textDecorationLine": undefined,
      }
    }
    testID="my-text"
  >
    hello world
  </Text>
</RNCSafeAreaProvider>
`;
